packages/teams-js/src/public/pages.ts

```diff
@@ -69,6 +70,24 @@ export namespace pages {
     });
   }
 
+  export function navigateToApp(params: NavigateToAppParams): Promise<void> {
+    return new Promise<void>(resolve => {
+      ensureInitialized(
+        FrameContexts.content,
+        FrameContexts.sidePanel,
+        FrameContexts.settings,
+        FrameContexts.task,
+        FrameContexts.stage,
+        FrameContexts.meetingStage,
+      );
+      if (runtime.isLegacyTeams) {
+        resolve(send('executeDeepLink', createTeamsAppLink(params)));
```

Comment:
I still don't understand why we're resolving with a `Promise<void>` as parameter. Wouldn't that result in a `Promise<Promise<void>>`? This seems weird, not just here, but in all the places we do that.

Reply:
Nope! The `resolve` method will unwrap the Promise if the method you call returns one. It's a common way to create a Promise that then calls other async code.